### PR TITLE
KAFKA-9891: Add integration test to replicate standby task failover

### PR DESCRIPTION
**Pull request by abbccdda**
This integration test is trying to mimic the scenario where the active task writes an invalid record while the standby task replicates it under at_least_once. For EOS, it should not happen because the data is not committed.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskFailOverIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, appId);
+        streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        streamsConfiguration.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
+        streamsConfiguration.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, 1);
+        streamsConfiguration.put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, numThreads);
+        streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG, stateDirPath);
+        streamsConfiguration.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, eosConfig);
+        streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.Integer().getClass());
+        streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.Integer().getClass());
+        // Set commit interval long to avoid actually committing the record.
```
Comment (mateuszjadczykDna) [inline streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskFailOverIntegrationTest.java:276]: why is it actually needed?

File: streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskFailOverIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, Integer.MAX_VALUE);
+        streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        streamsConfiguration.put(ProducerConfig.LINGER_MS_CONFIG, 0);
+
```
Comment (mateuszjadczykDna) [inline streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskFailOverIntegrationTest.java:280]: I've noticed there's only a single rebalance, and we need at least 3. maybe set 
`        streamsConfiguration.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, 1000);
        streamsConfiguration.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, 500);`
or similar

File: streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskFailOverIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    @Test
+    public void testStandbyTaskFailOver() throws Exception {
+
+        CountDownLatch waitPoisonRecordReplication = new CountDownLatch(1);
+
+        streamInstanceOne = getStreamInstance(1, waitPoisonRecordReplication);
+        streamInstanceTwo = getStreamInstance(2, null);
+
+        CountDownLatch threadDeaths = new CountDownLatch(3);
```
Comment (mateuszjadczykDna) [inline streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskFailOverIntegrationTest.java:124]: eventually all 4 threads should be down

File: streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskFailOverIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+                    }
+                }
+                return false;
+            }, "Did not see poison key replicated to instance two");
+        } else {
+            // Wait sufficient time to make sure the data is not replicated.
+            Thread.sleep(3000);
+        }
+
+        threadDeaths.await(15, TimeUnit.SECONDS);
```
Comment (mateuszjadczykDna) [inline streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskFailOverIntegrationTest.java]: assert at the end that all threads went down?

**Comment by mjsax** [comment]
The other PRs got merged. Closing this PR.

**Pull request closed without merging**